app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue

```diff
@@ -133,7 +144,11 @@ function toggleDropdownWidget() {
 }
 
 function onClick(index: number, keepOpen: boolean) {
-  selectedIndex.value = index
+  if (index < customTags.value.length) {
+    customTags.value[index]!.onClick()
+  } else {
+    selectedIndex.value = index
+  }
```

Comment:
This code introduces unnecessary tag ordering invariant, but it can be easily avoided.
```suggestion
  if (tags.value[index]?.onClick?.() ?? true) selectedIndex.value = index
```

With that, I think the whole `customTags` computed can be inlined into `tags`, and maybe we don't need a separate `CustomTag` type at all here. The `interface Tag` could simply have additional field `onClick?(): boolean | void`. We also get for free the ability to `return true` to keep the default selection logic still working, though it's not really that relevant.

Also then, the `CustomDropdownItemsKey` input value could simply accept an array of `Tag`s, again making `CustomDropdownItem` type unnecessary.

At this point, it's also worth to rename `Tag` interface to something more appropriate, like `DropdownItem`.